token/client/src/token.rs

```diff
@@ -51,6 +51,8 @@ pub enum TokenError {
     AccountInvalidMint,
     #[error("proof error: {0}")]
     Proof(ProofError),
+    #[error("illegal amount")]
+    IllegalAmount,
```

Comment:
Perhaps an error that's more precise here, as the amount isn't really "illegal". `MaximumDepositTransferAmountExceeded`?

Reply:
Yes, that is a more decryptive and much better name indeed.